### PR TITLE
Initial classes

### PR DESCRIPTION
**Pull request by mecha**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,13 +12,23 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "require": {
-        "php": "^5.4 | ^7.0"
+        "php": "^5.4 | ^7.0",
+        "dhii/factory-interface": "0.1-alpha1 | 0.1-alpha2",
+        "dhii/callback-abstract": "0.1-alpha2",
+        "dhii/invocable-interface": "0.1-alpha1",
+        "dhii/invocable-base": "dev-task/initial-classes",
```
Comment (XedinUnknown) [inline composer.json]: You can now depend on [`dhii/invocable-base@0.1-alpha1`][dhii/invocable-base@0.1-alpha1].

[dhii/invocable-base@0.1-alpha1]: https://github.com/Dhii/invocable-base/releases/tag/v0.1-alpha1

File: composer.json [hunk 2]
```diff
@@ -28,5 +38,16 @@
     "scripts": {
         "test": "phpunit",
         "csfix": "php-cs-fixer fix -vvv"
+    },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "http://github.com/dhii/invocable-base"
```
Comment (XedinUnknown) [inline composer.json]: Now published [here](https://packagist.org/packages/dhii/invocable-base).

File: composer.json [hunk 3]
```diff
@@ -28,5 +38,16 @@
...
+        }
+    ],
+    "config": {
+        "platform": {
+            "php": "5.4.0"
```
Comment (XedinUnknown) [inline composer.json]: This looks extra.

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

**Review by XedinUnknown — APPROVED** [review]

**Comment by XedinUnknown** [comment]
@mecha, ready for merge?

**Comment by mecha** [comment]
@XedinUnknown Let's see if it builds successfully first

File: src/SubjectConfigAwareTrait.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+     * @var array|ArrayAccess|stdClass|ContainerInterface
+     */
+    protected $subjectConfig;
+
+    /**
+     * Retrieves the subject factory configuration associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @return array|ArrayAccess|ContainerInterface|stdClass The subject factory configuration, if any.
```
Comment (XedinUnknown) [inline src/SubjectConfigAwareTrait.php]: Cannot be null?

File: src/FactoryAwareTrait.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+ * @since [*next-version*]
+ */
+trait FactoryAwareTrait
+{
+    /**
+     * The factory associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @var FactoryInterface
```
Comment (XedinUnknown) [inline src/FactoryAwareTrait.php]: Cannot be `null`?

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

File: src/FactoryAwareTrait.php (hunk 5 above)
Comment (mecha) [reply src/FactoryAwareTrait.php]: Can be. Nice catch.

File: src/SubjectConfigAwareTrait.php (hunk 4 above)
Comment (mecha) [reply src/SubjectConfigAwareTrait.php]: Can be. Nice catch.

**Review by XedinUnknown — APPROVED** [review]

**Pull request merged**